bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

```diff
@@ -45,14 +50,14 @@
             new Capabilities(7).primary()                         .secondary().text("Top Down Bottom Up"),
             new Capabilities(8).primary()                                     .text("Duolite Lift"),
             new Capabilities(9).primary().tiltAnywhere()                      .text("Duolite Lift and Tilt 90°"),
-        // @formatter:on
-            new Capabilities() };
+    // @formatter:on
+            new Capabilities()).stream().collect(Collectors.toMap(Capabilities::getValue, Function.identity()));
```

Comment:
I believe this can be simplified and optimized (by avoiding some unneeded conversions) into something like:

```
    private static final Map<Integer, Capabilities> CAPABILITIES_DATABASE = Map.ofEntries(
    // @formatter:off
            Map.entry(0, new Capabilities(0).primary().tiltOnClosed()                      .text("Bottom Up")),
            Map.entry(1, new Capabilities(1).primary().tiltAnywhere()                      .text("Bottom Up Tilt 90°")),
[...]
    // @formatter:on
            );
```

Reply:
Agreed. But why do you want a human to refactor it, when you can do it with a computer? ;)

Reply:
In particular i want to avoid the risk that the map's key and the value's value may be different. It is anyway a static final field so any minor code overhead would be one off on loading the JAR..

Reply:
@andrewfg - hmm, damn, I feel a joke here, but not getting it. :)

Reply:
Well, problem is that the key is also part of the value. But maybe it doesn't need to be. You have two usages:
- **isCapabilitiesInDatabase:** Just checks existence, key not needed.
- **getCapabilities:** Returns Capabilities from key. Which means caller already has the key, so should not be needed.

Reply:
It's not a simple as that. In some places it uses 'naked' Capabilities instances and/or 'naked' Type instances (ones which are explicitly not in the Map) that have an intValue of -1, and a specific text, to indicate an error. So it is not possible to  relocate the Value intValue to the Map Key. If we are to use a Map, then we must to duplicate the Map key and Value intValue. (That is the original reason why I did not use a Map). If you insist on having a Map, the I cannot accept your proposed method of manual Map entry creation because it introduces the risk of creating a entries where the Map Key and the Value intValue are not equal. The advantage of the `Arrays.asList().stream().collect(Collectors.toMap(Capabilities::getValue, Function.identity()))` technique, is that Value intValue is declared just once, and the class constructor creates the Map Key directly from it.